In [1]:
# @title
# ==========================================
# CONFIGURACIÓN DEL ENTORNO (EJECUTAR PRIMERO)
# ==========================================
import pandas as pd
import sqlite3

# 1) URLs de archivos crudos (Raw) en GitHub
urls = {
    'clientes': "https://raw.githubusercontent.com/vbroosing/prueba-tecnica/refs/heads/main/clientes.csv",
    'ventas': "https://raw.githubusercontent.com/vbroosing/prueba-tecnica/refs/heads/main/ventas.csv",
    'trabajadores': "https://raw.githubusercontent.com/vbroosing/prueba-tecnica/refs/heads/main/trabajadores.csv",
    'cargos': "https://raw.githubusercontent.com/vbroosing/prueba-tecnica/refs/heads/main/cargos.csv",
    'areas': "https://raw.githubusercontent.com/vbroosing/prueba-tecnica/refs/heads/main/areas.csv"
}

print("Preparando base de datos 'techstore.db'.. .")

# 2) Crear la conexión y la base de datos
conn = sqlite3.connect('techstore.db')

# 3) Leer cada CSV e insertarlo como tabla en la base de datos
for tabla, url in urls.items():
    try:
        df = pd.read_csv(url, sep=None, engine='python')
        df.to_sql(tabla, conn, index=False, if_exists='replace')
        print(f"Tabla '{tabla}' cargada exitosamente.")
    except Exception as e:
        print(f"Error al cargar la tabla '{tabla}': {e}")

conn.close()
print("¡Entorno listo! La base de datos relacional está disponible para ser consultada.")

Preparando base de datos 'techstore.db'.. .
Tabla 'clientes' cargada exitosamente.
Tabla 'ventas' cargada exitosamente.
Tabla 'trabajadores' cargada exitosamente.
Tabla 'cargos' cargada exitosamente.
Tabla 'areas' cargada exitosamente.
¡Entorno listo! La base de datos relacional está disponible para ser consultada.


In [2]:
# @title
import sqlite3
import pandas as pd

conn = sqlite3.connect('techstore.db')

print("Conectado correctamente")

Conectado correctamente


#Parte 1 Ejercicios

In [3]:
pd.read_sql("""
SELECT name
FROM sqlite_master
WHERE type='table'
""", conn)

,name
0,clientes
1,ventas
2,trabajadores
3,cargos
4,areas


In [4]:
pd.read_sql(
"""
SELECT *
FROM ventas
LIMIT 10
""",
conn
)

,id,id_cliente,id_vendedor,dte,total_dte,estado,descripcion
0,1,191,15,10001,650631,Entregado,"2x Disco Duro Externo 2TB, 3x Mouse Ergonómico..."
1,2,109,3,10004,903198,Entregado,"2x Escáner de Documentos Portátil, 1x Mouse Er..."
2,3,76,10,10007,231213,Entregado,"2x Escáner de Documentos Portátil, 2x Memoria ..."
3,4,158,14,10009,752982,Sin entregar,"3x Licencia Software Antivirus 1 Año, 3x Lapto..."
4,5,132,6,10011,36388,Entregado,3x Mouse Ergonómico Inalámbrico
5,6,201,15,10014,706762,Entregado,1x Mouse Ergonómico Inalámbrico
6,7,200,12,10015,319835,Entregado,"3x Memoria RAM 16GB DDR4, 1x Escáner de Docume..."
7,8,243,12,10017,306798,Entregado,"1x Silla de Escritorio Ergonómica, 3x Monitor ..."
8,9,81,9,10019,444584,Entregado,"3x Monitor LED 24' Full HD, 1x Laptop Corporat..."
9,10,133,13,10021,30402,Sin entregar,2x Impresora Láser Multifuncional


In [5]:
pd.read_sql(
"""
SELECT *
FROM trabajadores
LIMIT 10
""",
conn
)

,id,id_cargo,nombre,apellido,rut
0,1,3,Juan,Donoso,14108603-0
1,2,2,Pedro,Donoso,21353133-6
2,3,2,Matías,Sepúlveda,10499914-K
3,4,2,Francisco,Silva,19416129-8
4,5,2,Felipe,Sepúlveda,17536477-3
5,6,2,María,Lara,21712835-8
6,7,2,Andrea,Martínez,13612365-3
7,8,3,Javiera,Rojas,11622631-6
8,9,3,Natalia,Cárcamo,10728977-1
9,10,3,Bárbara,Díaz,11322047-3


In [6]:
pd.read_sql(
"""
SELECT *
FROM cargos
""",
conn
)

,id,nombre_cargo,descripcion,id_area
0,1,Gerente Comercial,Liderar las estrategias de venta y cumplimient...,1
1,2,Ejecutivo de Ventas Senior,Venta directa y gestión de grandes cuentas de ...,1
2,3,Ejecutivo de Ventas Junior,Atención de prospectos y cierre de ventas de c...,1
3,4,Jefe de Logística,Coordinar la cadena de suministro y despacho a...,2
4,5,Operario de Bodega,"Preparación de pedidos, recepción de mercaderí...",2
5,6,Ingeniero de Soporte TI,"Resolver incidencias de hardware, software y r...",3
6,7,Desarrollador Full Stack,Mantenimiento y desarrollo de las plataformas ...,3
7,8,Analista de RRHH,"Procesamiento de remuneraciones, contratos y g...",4
8,9,Contador General,"Declaraciones de impuestos, balances comercial...",5
9,10,Analista Financiero,"Control de presupuestos, flujo de caja y relac...",5


In [7]:
pd.read_sql(
"""
SELECT *
FROM areas
""",
conn
)

,id,nombre_area,descripcion
0,1,Ventas y Comercial,Área encargada de la comercialización de produ...
1,2,Operaciones y Logística,"Gestión de inventario, bodegaje y despacho de ..."
2,3,Tecnología y Soporte,Mantenimiento de infraestructura tecnológica y...
3,4,Recursos Humanos,"Gestión del talento, contratación y bienestar ..."
4,5,Finanzas y Contabilidad,"Administración de recursos financieros, factur..."


In [8]:
pd.read_sql(
"""
SELECT descripcion
FROM ventas
LIMIT 20
""",
conn
)

,descripcion
0,"2x Disco Duro Externo 2TB, 3x Mouse Ergonómico..."
1,"2x Escáner de Documentos Portátil, 1x Mouse Er..."
2,"2x Escáner de Documentos Portátil, 2x Memoria ..."
3,"3x Licencia Software Antivirus 1 Año, 3x Lapto..."
4,3x Mouse Ergonómico Inalámbrico
5,1x Mouse Ergonómico Inalámbrico
6,"3x Memoria RAM 16GB DDR4, 1x Escáner de Docume..."
7,"1x Silla de Escritorio Ergonómica, 3x Monitor ..."
8,"3x Monitor LED 24' Full HD, 1x Laptop Corporat..."
9,2x Impresora Láser Multifuncional


In [9]:
trabajadores = pd.read_sql("""
SELECT *
FROM trabajadores
LIMIT 20
""", conn)

trabajadores

,id,id_cargo,nombre,apellido,rut
0,1,3,Juan,Donoso,14108603-0
1,2,2,Pedro,Donoso,21353133-6
2,3,2,Matías,Sepúlveda,10499914-K
3,4,2,Francisco,Silva,19416129-8
4,5,2,Felipe,Sepúlveda,17536477-3
5,6,2,María,Lara,21712835-8
6,7,2,Andrea,Martínez,13612365-3
7,8,3,Javiera,Rojas,11622631-6
8,9,3,Natalia,Cárcamo,10728977-1
9,10,3,Bárbara,Díaz,11322047-3


In [10]:
trabajadores.columns

Index(['id', 'id_cargo', 'nombre', 'apellido', 'rut'], dtype='object')

In [11]:
ventas = pd.read_sql("""
SELECT *
FROM ventas
LIMIT 20
""", conn)

ventas.columns

Index(['id', 'id_cliente', 'id_vendedor', 'dte', 'total_dte', 'estado',
       'descripcion'],
      dtype='object')

In [12]:
pd.read_sql("""
SELECT SUM(total_dte)
FROM ventas
WHERE estado = 'Entregado'
""", conn)

,SUM(total_dte)
0,477191018


In [13]:
pd.read_sql("""
SELECT SUM(total_dte)
FROM ventas
""", conn)

,SUM(total_dte)
0,588673484


In [14]:
pd.read_sql("""
SELECT
    SUM(total_dte) AS ventas_entregadas,
    ROUND(
        SUM(total_dte) * 100.0 /
        (SELECT SUM(total_dte) FROM ventas),
        2
    ) AS porcentaje_del_total
FROM ventas
WHERE estado = 'Entregado'
""", conn)

,ventas_entregadas,porcentaje_del_total
0,477191018,81.06


**Ingresos por ventas entregadas: 477.191.018.**

**Estas representan el 81,06% del total de ingresos registrados en la base de datos.**

In [15]:
pd.read_sql("""
SELECT
    c.id,
    c.rut,
    SUM(v.total_dte) AS gasto_total
FROM clientes c
JOIN ventas v
    ON c.id = v.id_cliente
GROUP BY c.id, c.rut
ORDER BY gasto_total DESC
LIMIT 5
""", conn)

,id,rut,gasto_total
0,82,21866669-8,7125123
1,81,10212267-4,5663000
2,94,11686548-3,5621201
3,145,14605016-6,5604037
4,48,19301497-6,5106230


In [16]:
pd.read_sql("""
SELECT
    id_cliente,
    dte
FROM ventas
ORDER BY id_cliente, dte
LIMIT 30
""", conn)

,id_cliente,dte
0,1,10461
1,1,10831
2,1,11350
3,1,12033
4,1,12206
5,2,10821
6,2,10986
7,2,11721
8,2,11918
9,3,10763


In [17]:
pd.read_sql("""
SELECT
    id_cliente,
    COUNT(*) as cantidad_compras
FROM ventas
GROUP BY id_cliente
ORDER BY cantidad_compras DESC
LIMIT 10
""", conn)

,id_cliente,cantidad_compras
0,147,11
1,148,10
2,84,10
3,82,10
4,81,10
5,73,10
6,48,10
7,30,10
8,224,9
9,180,9


In [18]:
ventas_df = pd.read_sql("""
SELECT *
FROM ventas
""", conn)

ventas_df.head()

,id,id_cliente,id_vendedor,dte,total_dte,estado,descripcion
0,1,191,15,10001,650631,Entregado,"2x Disco Duro Externo 2TB, 3x Mouse Ergonómico..."
1,2,109,3,10004,903198,Entregado,"2x Escáner de Documentos Portátil, 1x Mouse Er..."
2,3,76,10,10007,231213,Entregado,"2x Escáner de Documentos Portátil, 2x Memoria ..."
3,4,158,14,10009,752982,Sin entregar,"3x Licencia Software Antivirus 1 Año, 3x Lapto..."
4,5,132,6,10011,36388,Entregado,3x Mouse Ergonómico Inalámbrico


In [19]:
ventas_df[ventas_df["id_cliente"] == 2][["id_cliente","dte"]]

,id_cliente,dte
393,2,10821
475,2,10986
844,2,11721
947,2,11918


In [20]:
ventas_df[ventas_df["id_cliente"] == 1][["id_cliente","dte"]]

,id_cliente,dte
217,1,10461
398,1,10831
657,1,11350
1006,1,12033
1092,1,12206


In [21]:
ventas_df[['id','descripcion']].head(20)

,id,descripcion
0,1,"2x Disco Duro Externo 2TB, 3x Mouse Ergonómico..."
1,2,"2x Escáner de Documentos Portátil, 1x Mouse Er..."
2,3,"2x Escáner de Documentos Portátil, 2x Memoria ..."
3,4,"3x Licencia Software Antivirus 1 Año, 3x Lapto..."
4,5,3x Mouse Ergonómico Inalámbrico
5,6,1x Mouse Ergonómico Inalámbrico
6,7,"3x Memoria RAM 16GB DDR4, 1x Escáner de Docume..."
7,8,"1x Silla de Escritorio Ergonómica, 3x Monitor ..."
8,9,"3x Monitor LED 24' Full HD, 1x Laptop Corporat..."
9,10,2x Impresora Láser Multifuncional


In [22]:
import pandas as pd
import re

productos = []

for _, fila in ventas_df.iterrows():

    venta_id = fila['id']

    items = fila['descripcion'].split(',')

    for item in items:

        item = item.strip()

        match = re.match(r'(\d+)x\s+(.*)', item)

        if match:

            cantidad = int(match.group(1))
            producto = match.group(2)

            productos.append([
                venta_id,
                producto,
                cantidad
            ])

productos_df = pd.DataFrame(
    productos,
    columns=[
        'id_venta',
        'producto',
        'unidades'
    ]
)

top_productos = (
    productos_df
    .groupby('producto', as_index=False)
    ['unidades']
    .sum()
    .sort_values('unidades', ascending=False)
    .head(10)
)

top_productos

,producto,unidades
2,Escáner de Documentos Portátil,551
9,Mouse Ergonómico Inalámbrico,532
5,Laptop Corporate 14',504
3,Hub USB-C 8 en 1,502
7,Memoria RAM 16GB DDR4,492
1,Disco Duro Externo 2TB,492
6,Licencia Software Antivirus 1 Año,486
10,Silla de Escritorio Ergonómica,479
11,Teclado Mecánico RGB,475
0,Auriculares con Micrófono USB,471


###### Productos más vendidos

Se procesó la columna descripcion para separar las cantidades y nombres de productos contenidos en cada venta.

Posteriormente se consolidaron las unidades vendidas por producto y se obtuvo el ranking de los 10 productos con mayor volumen de ventas.

Este análisis permite identificar los productos con mayor rotación dentro del catálogo.

###Ventas Totales Entragadas

Ingresos por ventas entregadas: 477.191.018

Porcentaje sobre el total de ventas:
81,06%

Top Clientes

Se identificaron los 5 clientes con mayor gasto histórico mediante JOIN entre clientes y ventas, agrupando por cliente y sumando total_dte.

In [23]:
ventas_ret = ventas_df[['id_cliente','dte']].copy()

ventas_ret = ventas_ret.sort_values(
    ['id_cliente','dte']
)

In [24]:
ventas_ret['siguiente_compra'] = (
    ventas_ret.groupby('id_cliente')['dte']
    .shift(-1)
)

In [25]:
ventas_ret['diferencia_dte'] = (
    ventas_ret['siguiente_compra']
    - ventas_ret['dte']
)

ventas_ret.head(20)

,id_cliente,dte,siguiente_compra,diferencia_dte
217,1,10461,10831.0,370.0
398,1,10831,11350.0,519.0
657,1,11350,12033.0,683.0
1006,1,12033,12206.0,173.0
1092,1,12206,NaN,NaN
393,2,10821,10986.0,165.0
475,2,10986,11721.0,735.0
844,2,11721,11918.0,197.0
947,2,11918,NaN,NaN
364,3,10763,10827.0,64.0


In [26]:
clientes_con_retencion = (
    ventas_ret[ventas_ret['diferencia_dte'] <= 200]
    ['id_cliente']
    .nunique()
)

total_clientes = ventas_ret['id_cliente'].nunique()

porcentaje_retencion = (
    clientes_con_retencion
    / total_clientes
    * 100
)

print("Clientes con recompra:", clientes_con_retencion)
print("Total clientes:", total_clientes)
print("Retención:", round(porcentaje_retencion, 2), "%")

Clientes con recompra: 173
Total clientes: 248
Retención: 69.76 %


**Se consideró retenido un cliente cuando al menos una compra fue seguida por otra compra dentro de los 200 DTE siguientes.**

#Parte 2 Ejercicios

In [27]:
clientes_df = pd.read_sql("""
SELECT *
FROM clientes
""", conn)

clientes_df.head()

,id,rut,nombre,segundo_nombre,apellido,segundo_apellido,edad,fecha_nacimiento
0,1,12154051-7,Carlos,Javier,Medina,Rojas,55.0,13-04-1971
1,2,13553384-K,Bárbara,Carolina,Martínez,Valenzuela,29.0,17-12-1997
2,3,17350099-8,Rodrigo,Alejandro,Vergara,Araya,33.0,02-07-1993
3,4,10352896-8,Javiera,Valentina,Silva,Rojas,36.0,01-05-1990
4,5,13841005-6,Matías,Manuel,González,Rojas,61.0,13-04-1965


In [28]:
print("Valores nulos por columna:")
print(clientes_df.isnull().sum())

print("\nFilas duplicadas:")
print(clientes_df.duplicated().sum())

Valores nulos por columna:
id                  0
rut                 0
nombre              0
segundo_nombre      0
apellido            0
segundo_apellido    0
edad                4
fecha_nacimiento    0
dtype: int64

Filas duplicadas:
0


### Limpieza de datos

Se encontraron 4 valores nulos en la columna edad.

No se encontraron filas duplicadas.

Como la cantidad de valores faltantes es baja, los dejé identificados para revisarlos según el análisis que se quiera realizar posteriormente.

In [29]:
ventas_df.nlargest(10, 'total_dte')[[
    'id',
    'id_cliente',
    'total_dte'
]]

,id,id_cliente,total_dte
320,321,75,999825
483,484,40,998087
627,628,45,996482
661,662,182,996396
458,459,144,996276
114,115,201,995217
734,735,110,995031
442,443,106,994428
178,179,163,994032
207,208,72,992339


### Top 10 montos de venta
Se identificaron las 10 ventas con mayor valor de total_dte.

Los montos más altos se encuentran cercanos al millón, siendo la venta con id 321 la de mayor valor registrada con 999.825.

In [40]:
ventas_df.groupby('id_cliente').size() \
    .reset_index(name='cantidad_compras') \
    .sort_values('cantidad_compras', ascending=False) \
    .head(3)

,id_cliente,cantidad_compras
145,147,11
46,48,10
82,84,10


###Clientes con mayor cantidad de compras
Se identificaron los clientes con mayor volumen de transacciones.

El cliente 147 registra la mayor cantidad de compras con 11 transacciones. Los clientes 48 y 84 le siguen con 10 compras cada uno.

In [31]:
cargos_df = pd.read_sql("""
SELECT *
FROM cargos
""", conn)

cargos_df

,id,nombre_cargo,descripcion,id_area
0,1,Gerente Comercial,Liderar las estrategias de venta y cumplimient...,1
1,2,Ejecutivo de Ventas Senior,Venta directa y gestión de grandes cuentas de ...,1
2,3,Ejecutivo de Ventas Junior,Atención de prospectos y cierre de ventas de c...,1
3,4,Jefe de Logística,Coordinar la cadena de suministro y despacho a...,2
4,5,Operario de Bodega,"Preparación de pedidos, recepción de mercaderí...",2
5,6,Ingeniero de Soporte TI,"Resolver incidencias de hardware, software y r...",3
6,7,Desarrollador Full Stack,Mantenimiento y desarrollo de las plataformas ...,3
7,8,Analista de RRHH,"Procesamiento de remuneraciones, contratos y g...",4
8,9,Contador General,"Declaraciones de impuestos, balances comercial...",5
9,10,Analista Financiero,"Control de presupuestos, flujo de caja y relac...",5


In [32]:
trabajadores_df = pd.read_sql("""
SELECT *
FROM trabajadores
""", conn)

trabajadores_df.head()

,id,id_cargo,nombre,apellido,rut
0,1,3,Juan,Donoso,14108603-0
1,2,2,Pedro,Donoso,21353133-6
2,3,2,Matías,Sepúlveda,10499914-K
3,4,2,Francisco,Silva,19416129-8
4,5,2,Felipe,Sepúlveda,17536477-3


In [33]:
areas_df = pd.read_sql("""
SELECT *
FROM areas
""", conn)

In [34]:
top_vendedores = pd.read_sql("""
SELECT
    t.id,
    t.nombre,
    t.apellido,
    c.nombre_cargo,
    SUM(v.total_dte) AS monto_total
FROM ventas v
JOIN trabajadores t
    ON v.id_vendedor = t.id
JOIN cargos c
    ON t.id_cargo = c.id
WHERE t.id_cargo IN (1,2,3)
GROUP BY
    t.id,
    t.nombre,
    t.apellido,
    c.nombre_cargo
ORDER BY monto_total DESC
LIMIT 3
""", conn)

top_vendedores

,id,nombre,apellido,nombre_cargo,monto_total
0,5,Felipe,Sepúlveda,Ejecutivo de Ventas Senior,40264872
1,13,Manuel,Cortés,Ejecutivo de Ventas Senior,38625479
2,7,Andrea,Martínez,Ejecutivo de Ventas Senior,36945022


### Análisis de vendedores

Se realizó una consulta para identificar a los vendedores con mayor monto acumulado en ventas.

Los tres vendedores con mejor desempeño fueron:

- Felipe Sepúlveda: $40.264.872

- Manuel Cortés: $38.625.479

- Andrea Martínez: $36.945.022

En los tres casos el cargo corresponde a Ejecutivo de Ventas Senior, lo que sugiere que este perfil concentra gran parte de las ventas realizadas en la empresa.

In [35]:
import pandas as pd
import re

productos = []

for _, fila in ventas_df.iterrows():

    venta_id = fila['id']

    items = fila['descripcion'].split(',')

    for item in items:

        item = item.strip()

        match = re.match(r'(\d+)x\s+(.*)', item)

        if match:

            cantidad = int(match.group(1))
            producto = match.group(2)

            productos.append([
                venta_id,
                producto,
                cantidad
            ])

productos_df = pd.DataFrame(
    productos,
    columns=[
        'id_venta',
        'producto',
        'unidades'
    ]
)

top_productos = (
    productos_df
    .groupby('producto', as_index=False)
    ['unidades']
    .sum()
    .sort_values('unidades', ascending=False)
    .head(10)
)

top_productos

,producto,unidades
2,Escáner de Documentos Portátil,551
9,Mouse Ergonómico Inalámbrico,532
5,Laptop Corporate 14',504
3,Hub USB-C 8 en 1,502
7,Memoria RAM 16GB DDR4,492
1,Disco Duro Externo 2TB,492
6,Licencia Software Antivirus 1 Año,486
10,Silla de Escritorio Ergonómica,479
11,Teclado Mecánico RGB,475
0,Auriculares con Micrófono USB,471


### Top 10 productos más vendidos

Luego de procesar la columna `descripcion`, se obtuvo el ranking de productos más vendidos según la cantidad total de unidades comercializadas.

El producto con mayor volumen de venta fue "Escáner de Documentos Portátil" con 551 unidades, seguido por "Mouse Ergonómico Inalámbrico" con 532 unidades.

Se observa que la demanda se concentra principalmente en accesorios y equipamiento tecnológico de oficina.

In [36]:
ventas_df["descripcion"].str.split(",").explode().value_counts().head(10)

,count
descripcion,
1x Silla de Escritorio Ergonómica,63
1x Auriculares con Micrófono USB,62
2x Monitor LED 24' Full HD,61
1x Laptop Corporate 14',60
1x Licencia Software Antivirus 1 Año,59
3x Mouse Ergonómico Inalámbrico,59
2x Escáner de Documentos Portátil,58
2x Licencia Software Antivirus 1 Año,57
2x Auriculares con Micrófono USB,55


### Análisis de productos

Al revisar las descripciones de venta se observa que algunos productos aparecen de forma recurrente en distintas transacciones.

Entre los productos más frecuentes se encuentran:

- Mouse Ergonómico Inalámbrico
- Monitor LED 24" Full HD
- Escáner de Documentos Portátil
- Memoria RAM 16GB DDR4
- Laptop Corporativa

Esto indica que estos artículos tienen una alta rotación y representan una parte importante de las ventas realizadas.

#Parte 3 Ejercicios (Google Sheets)

En esta etapa se importaron los archivos CSV generados en Python a Google Sheets para complementar el análisis. Se trabajó con una muestra de las primeras 100 ventas, utilizando funciones BUSCARX para relacionar información de clientes, cargos y áreas. Además, se creó una tabla dinámica para analizar las ventas entregadas por área y se clasificaron las ventas como Ticket Alto o Ticket Bajo utilizando el promedio de ventas como referencia.

In [37]:
clientes_df.to_csv("clientes.csv", index=False)
ventas_df.to_csv("ventas.csv", index=False)
trabajadores_df.to_csv("trabajadores.csv", index=False)
cargos_df.to_csv("cargos.csv", index=False)
areas_df.to_csv("areas.csv", index=False)

In [38]:
from google.colab import files

files.download("clientes.csv")
files.download("ventas.csv")
files.download("trabajadores.csv")
files.download("cargos.csv")
files.download("areas.csv")

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

# Conclusiones

- El 81,06% del monto vendido fue entregado correctamente.
- Se identificaron 4 valores nulos en la columna edad.
- No se encontraron registros duplicados.
- El cliente con mayor gasto acumulado supera los $7 millones.
- La tasa de retención de clientes alcanza un 69,76%.
- Los ejecutivos de ventas senior concentran los mayores montos de venta.
- Existen productos que aparecen de forma recurrente en las ventas, indicando una alta demanda.

####Anexos

#**Contexto de la Prueba: E-Commerce "TechStore"**

*Escenario: Eres el nuevo analista de datos de TechStore. El equipo ha notado una fluctuación en los ingresos durante el último trimestre y necesita entender qué está pasando.*

**Instrucciones Iniciales:**

1.   La prueba tiene una duración estimada de 45 mins, pero tendrás 72 horas para enviar tus resultados.
2.	Luego de terminar tus ejercicios envíame el enlace en compartir para tu entorno donde fue desarrollado.
3.	Cualquier archivo externo generado debes subirlo y compartir el enlace en una nueva hoja de google colab, en caso de usar hojas de cálculo de Google debes copiar el enlace en otra hoja de google colab, recuerda titular dicho enlace para saber que iré a mirar.
4.	Dentro de la prueba, ve a Archivo > Guardar una copia en Drive para tener tu propia versión editable de esta prueba.
5.	Ejecuta la primera celda de este cuaderno. Esto generará automáticamente una base de datos SQLite llamada techstore.db en este entorno.
6.	El diccionario de datos disponible en la base de datos es:
*   clientes: id, rut, nombre, segundo_nombre, apellido, segundo_apellido, edad, fecha_nacimiento
*   ventas: id, id_cliente, id_vendedor, dte, total_dte, estado, descripcion
*   trabajadores: id, id_cargo, nombre, apellido, rut
*   cargos: id, nombre_cargo, descripcion, id_area
*   areas: id, nombre_area, descripcion

*disclaimer: Puedes usar IA, sin embargo, revisaré cada linea de código para validar la efectividad y también la totalidad del cumplimiento de las respuestas.*

**Parte 1: Conexión y Extracción (SQL & Python)**

*Conéctate a la base de datos techstore.db usando Python (por ejemplo, con la librería sqlite3 o sqlalchemy). Luego, resuelve las siguientes consultas ejecutando SQL puro dentro de tu código Python:*
1.   **Ventas Totales:** Escribe una consulta para calcular el ingreso total generado por órdenes con estado "Entregado" y el % del total de ventas que representa.
2.   **Top Clientes:** Encuentra los 5 usuarios que han gastado más dinero históricamente, mostrando su id, rut y el total_dte gastado.
3.   **Retención:** Calcula el porcentaje de usuarios que realizaron una segunda compra dentro de los 200 dte siguientes (si alguien compró el dte 11721 y compró nuevamente antes ó justamente en el dte 11921 es considerado).

**Parte 2: Análisis Exploratorio (Python)**

*Trae las tablas necesarias a DataFrames de Pandas y resuelve:*

1.	**Limpieza:** Identifica y muestra los valores nulos o duplicados en la tabla de clientes. Comenta tu criterio.
2.	**Métricas:** Identifica el top 10 de total_dte más altos, el top 3 de clientes por volumen de transacciones (cantidad de compras), y el top 3 de vendedores (id_vendedor) con mayor monto acumulado, solo considera vendedores, si hay otros cargos comerciales que han realizado ventas debes ignorarlos.
3.	**Volumen:** Crea un DataFrame del top 10 de productos mas vendidos ordenado de mayor a menor que muestre el id de la venta, nombre del producto y unidades vendidas (deberás procesar la columna descripcion).


**Parte 3: Lógica de Negocio (Hojas de Cálculo / Excel)**

*Para esta sección, deberás ingeniártelas para extraer/descargar los datos de este entorno (ej. generando archivos CSV desde tus DataFrames o desde la DB) e importarlos a Google Sheets o Excel, puedes utilizar la tecnología que estimes conveniente, lo importante es resolver en Excel.*

1.	**Consolidación**: En una hoja nueva, toma una muestra de las primeras 100 ventas según el dte asociado (a menor folio, el documento es más antiguo). Usa fórmulas de búsqueda (BUSCARV, BUSCARX o INDICE/COINCIDIR) para traer el nombre y apellido del cliente y el nombre_cargo del vendedor asociado a esa venta.
2.	**Tabla Dinámica:** Crea una tabla dinámica que muestre el monto total vendido por cada área de la empresa (nombre_area), filtrando solo el estado "Entregado".
3.	**Segmentación:** En tu hoja de ventas, crea la columna "Categoría de Ticket". Usa funciones lógicas para clasificar la venta como "Ticket Alto" (si el total_dte supera el promedio general) o "Ticket Bajo" (si es igual o inferior).


**Opcional (Puntos extra):**

**Presentación al Negocio (Dashboard + Insights)**

1.	**Visualización**: Conecta tus datos exportados a Power BI, Looker o Tableau. Crea un dashboard de una página con:


*   KPI de Ingresos Totales y Ticket Promedio.
*   Gráfico comparativo de rendimiento por categoría/producto.
*   Filtros interactivos.
2.	**Estrategia**: En un documento breve o en celdas de texto al final de este cuaderno, responde:

*   ¿Cuáles fueron tus hallazgos respecto a la fluctuación de ingresos?
*   Propón dos recomendaciones accionables para marketing/ventas.

**Control de versiones (Github):**

1.	**Creación del repositorio:** Entregar los scripts (SQL/Python) subidos a un repositorio propio de GitHub.
